### ISSUE TITLE
False negative for consider-using-ternary

### ISSUE THREAD

**Issue by zenlyj** [PyCQA/pylint#7954]
### Bug description

From my understanding, message should be emitted for consider-using-ternary when the following syntax is used for ternary operations:
`<CONDITIONAL_EXPRESSION> and <TRUE_VALUE> or <FALSE_VALUE>`

However, the relevant messages are not emitted when I execute the following script `bad.py`:

```
def print_max(x, y):
    maximum = x >= y and x or y    # usage of mentioned syntax
    print(maximum)

a, b = input(), input()
c = a >= b and a or b    # usage of mentioned syntax
```

### Command used

```shell
pylint bad.py
```

### Pylint output

```shell
************* Module bad
bad.py:1:0: C0114: Missing module docstring (missing-module-docstring)
bad.py:1:0: C0116: Missing function or method docstring (missing-function-docstring)
bad.py:1:14: C0103: Argument name "x" doesn't conform to snake_case naming style (invalid-name)
bad.py:1:17: C0103: Argument name "y" doesn't conform to snake_case naming style (invalid-name)
```

### Expected behavior

consider-using-ternary messages emitted on lines 2 and 6.

For line 2, there should be a suggestion to use the following syntax instead: `maximum = x if x >= y else y`
Similarly, suggestion for line 6 should be: `c = a if a >= b else b`.

### Pylint version

```shell
pylint 2.16.0-dev
astroid 2.12.13
Python 3.8.10 (default, Nov 14 2022, 12:59:47)
[GCC 9.4.0]
```

### OS / Environment

Ubuntu 20.04 LTS

**Comment by nickdrozd** [comment]
I would call this a new feature rather than a false negative, but it's a good idea either way. It's a special case of https://github.com/PyCQA/pylint/issues/2339.

**Comment by Pierre-Sassoulas** [comment]
Closing as duplicate of #2339
